Mathlib/Topology/DerivedSet.lean

```diff
@@ -64,6 +64,10 @@ lemma isClosed_iff_derivedSet_subset (A : Set X) : IsClosed A ↔ derivedSet A 
     rw [this, ← acc_principal_iff_cluster] at ha
     exact nh (h ha)
 
+lemma closure_eq_self_union_derivedSet (A : Set X) : closure A = A ∪ derivedSet A := by
+  show _ = {a | a ∈ A ∨ a ∈ derivedSet A}
+  simp [closure_eq_cluster_pts, clusterPt_principal]
```

Comment:
Does this work?

```suggestion
  ext
  simp [mem_closure_iff_clusterPt, clusterPt_principal]
```

It would be a little cleaner.

Reply:
nice, I pushed a very similar version with `ext`